libcontainer/configs/validate/validator.go

```diff
@@ -299,26 +299,18 @@ func checkIDMapMounts(config *configs.Config, m *configs.Mount) error {
 	if !m.IsIDMapped() {
 		return nil
 	}
-
 	if !m.IsBind() {
-		return fmt.Errorf("gidMappings/uidMappings is supported only for mounts with the option 'bind'")
+		return errors.New("id-mapped mounts are only supported for bind-mounts")
 	}
 	if config.RootlessEUID {
-		return fmt.Errorf("gidMappings/uidMappings is not supported when runc is being launched with EUID != 0, needs CAP_SYS_ADMIN on the runc parent's user namespace")
-	}
-	if len(config.UIDMappings) == 0 || len(config.GIDMappings) == 0 {
-		return fmt.Errorf("not yet supported to use gidMappings/uidMappings in a mount without also using a user namespace")
-	}
-	if !sameMapping(config.UIDMappings, m.UIDMappings) {
-		return fmt.Errorf("not yet supported for the mount uidMappings to be different than user namespace uidMapping")
-	}
-	if !sameMapping(config.GIDMappings, m.GIDMappings) {
-		return fmt.Errorf("not yet supported for the mount gidMappings to be different than user namespace gidMapping")
+		return errors.New("id-mapped mounts are not supported for rootless containers")
 	}
-	if !filepath.IsAbs(m.Source) {
-		return fmt.Errorf("mount source not absolute")
+	if len(m.UIDMappings) == 0 || len(m.GIDMappings) == 0 {
+		if m.Extensions&configs.EXT_ANY_IDMAP != 0 {
```

Comment:
I think we should not return two different errors. It should combine to one error like this:
`id-mapped mounts must have both uid and gid mappings specified or be used with a container using user namespaces`

Reply:
Oops. I added this error before I made `EXT_*IDMAP` be auto-set for all id-mapped mounts,  but now the second error will never be returned.

(FWIW, I think there was a benefit for having two errors -- the "or be used with a container using user namespaces" is confusing if the user hasn't specified `idmap` or `ridmap` but has instead only specified one of the mappings, triggering the error.)